nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQueryBatch.java

```diff
@@ -0,0 +1,269 @@
+        return ImmutableList.<PropertyDescriptor>builder()
+                .addAll(super.getSupportedPropertyDescriptors())
+                .add(DATASET)
+                .add(TABLE_NAME)
+                .add(TABLE_SCHEMA)
+                .add(SOURCE_TYPE)
+                .add(CREATE_DISPOSITION)
+                .add(WRITE_DISPOSITION)
+                .add(MAXBAD_RECORDS)
+                .add(IGNORE_UNKNOWN)
+                .build();
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+                .expressionLanguageSupported(true)
+                .dynamic(true)
+                .build();
+    }
+
+    @Override
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        super.onScheduled(context);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final Map<String, String> attributes = new HashMap<>();
+
+        final BigQuery bq = getCloudService();
+
+        final String projectId = context.getProperty(PROJECT_ID).evaluateAttributeExpressions().getValue();
+        final String dataset = context.getProperty(DATASET).evaluateAttributeExpressions(flowFile).getValue();
+        final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+
+        final TableId tableId;
+        if (StringUtils.isEmpty(projectId)) {
+            tableId = TableId.of(dataset, tableName);
+        } else {
+            tableId = TableId.of(projectId, dataset, tableName);
+        }
+
+        final String fileType = context.getProperty(SOURCE_TYPE).getValue();
+
+        String schemaString = context.getProperty(TABLE_SCHEMA).evaluateAttributeExpressions().getValue();
+        Schema schema = BqUtils.schemaFromString(schemaString);
+
+        WriteChannelConfiguration writeChannelConfiguration =
+                WriteChannelConfiguration.newBuilder(tableId)
+                        .setCreateDisposition(JobInfo.CreateDisposition.valueOf(context.getProperty(CREATE_DISPOSITION).getValue()))
+                        .setWriteDisposition(JobInfo.WriteDisposition.valueOf(context.getProperty(WRITE_DISPOSITION).getValue()))
+                        .setIgnoreUnknownValues(context.getProperty(IGNORE_UNKNOWN).asBoolean())
+                        .setMaxBadRecords(context.getProperty(MAXBAD_RECORDS).asInteger())
+                        .setSchema(schema)
+                        .setFormatOptions(FormatOptions.of(fileType))
+                        .build();
+
+        TableDataWriteChannel writer = bq.writer(writeChannelConfiguration);
```

Comment:
This should also be in the try/catch as it can raise exception which you should cause the processor to yield. Example:
````
com.google.cloud.bigquery.BigQueryException: 404 Not Found
Not Found
com.google.cloud.bigquery.BigQueryException: 404 Not Found
Not Found
        at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:99)
        at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.open(HttpBigQueryRpc.java:419)
        at com.google.cloud.bigquery.TableDataWriteChannel$2.call(TableDataWriteChannel.java:82)
        at com.google.cloud.bigquery.TableDataWriteChannel$2.call(TableDataWriteChannel.java:77)
        at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:89)
        at com.google.cloud.RetryHelper.run(RetryHelper.java:74)
        at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:51)
        at com.google.cloud.bigquery.TableDataWriteChannel.open(TableDataWriteChannel.java:76)
        at com.google.cloud.bigquery.TableDataWriteChannel.<init>(TableDataWriteChannel.java:41)
        at com.google.cloud.bigquery.BigQueryImpl.writer(BigQueryImpl.java:729)
        at com.google.cloud.bigquery.BigQueryImpl.writer(BigQueryImpl.java:720)
        at org.apache.nifi.processors.gcp.bigquery.PutBigQueryBatch.onTrigger(PutBigQueryBatch.java:211)
        at org.apache.nifi.processor.AbstractProcessor.onTrigger(AbstractProcessor.java:27)
        at org.apache.nifi.controller.StandardProcessorNode.onTrigger(StandardProcessorNode.java:1165)
        at org.apache.nifi.controller.tasks.ConnectableTask.invoke(ConnectableTask.java:203)
        at org.apache.nifi.controller.scheduling.TimerDrivenSchedulingAgent$1.run(TimerDrivenSchedulingAgent.java:117)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: com.google.api.client.http.HttpResponseException: 404 Not Found
Not Found
        at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1070)
        at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.open(HttpBigQueryRpc.java:416)
        ... 21 common frames omitted
````
````